src/main/java/Display.java

```diff
@@ -107,5 +103,45 @@ public String getOnlyLettersInput() {
     public void getUserName() {
         cli.present(Message.enterUserName());
     }
+
+    public void getStage() {
+        cli.present(Message.enterOpportunityStage());
+    }
+
+    public String getOpportunityStage() {
+        String stageChoice = getUserInputString();
+        while (!validator.opportunityStageChoice(stageChoice)) {
+            cli.present(Message.invalidMenuChoice());
+            stageChoice = getUserInputString();
+        }
+        return stageChoice;
+    }
+
+    private HashMap<Integer, ArrayList> formattedOpportunitiesData(List<Opportunity> opportunities) {
+        HashMap<Integer, ArrayList> formattedOpportunities = new HashMap<>();
+        for (Opportunity opp : opportunities) {
+            ArrayList<String> opportunityDetails = new ArrayList<>();
```

Comment:
Why do you make your List of Opportunity objects into a Hashmap containing List<String>? 
Can't you already use the details of each opportunity where you need them? A _list of opportunity details_ could be an _Opportunity_.